stream.go

```diff
@@ -338,9 +341,7 @@ NewStream:
 				if pub.Retry() > 0 {
 					stream.retryDelay.SetBaseDelay(time.Duration(pub.Retry()) * time.Millisecond)
 				}
-				if len(pub.Id()) > 0 {
-					stream.lastEventID = pub.Id()
-				}
+				stream.lastEventID = pub.lastEventID
```

Comment:
The reason for this change is that if we're only looking at `pub.Id()`, we can't distinguish between "this event had no `id:` field when we parsed it" and "this event had an `id:` field whose value was empty." In the former case, the last event ID should not be updated, but in the latter, it should; that is, you should be able to explicitly override the last ID to be "". That behavior is correctly implemented in the setting of `pub.lastEventID` by `Decoder`.

Reply:
Alternatively we could have provided another field whose meaning was "did this event have an ID explicitly set", or used a type with "nullable string" semantics such as `ldvalue.OptionalString` for the ID (or `*string`, but that has the disadvantage of being mutable and causing more heap allocations)... but, as with lastEventID, those would be hard to shoehorn into the existing interface.